packages/payment/src/models/payment-method-token.ts

```diff
@@ -0,0 +1,43 @@
+import {
+  BeforeCreate,
+  Entity,
+  OnInit,
+  PrimaryKey,
+  Property,
+} from "@mikro-orm/core"
+
+import { generateEntityId } from "@medusajs/utils"
+
+@Entity({ tableName: "payment_method_token" })
+export default class PaymentMethodToken {
+  @PrimaryKey({ columnType: "text" })
+  id: string
+
+  @Property({ columnType: "text" })
+  provider_id: string
+
+  @Property({ columnType: "jsonb", nullable: true })
+  data?: Record<string, unknown> | null
+
+  @Property({ columnType: "text" })
+  name: string
+
+  @Property({ columnType: "text", nullable: true })
+  type_detail?: string | null
+
+  @Property({ columnType: "text", nullable: true })
+  description_detail?: string | null
+
+  @Property({ columnType: "jsonb", nullable: true })
+  metadata?: Record<string, unknown> | null
```

Comment:
comment: I think we can put a default initializer on all those props. That way, they become optional by default wdyt? I will stop those comments until your feedback cc @carlos-r-l-rodrigues @olivermrbl 

```suggestion
  data: Record<string, unknown> | null = null
  
  @Property({ columnType: "text" })
  name: string

  @Property({ columnType: "text", nullable: true })
  type_detail: string | null = null

  @Property({ columnType: "text", nullable: true })
  description_detail: string | null = null

  @Property({ columnType: "jsonb", nullable: true })
  metadata: Record<string, unknown> | null = null
```

Reply:
in that case, should we use this convention across the new modules and does it make sense to change this now if we will be migrating to v6?

Reply:
we have an open discussion about it ahah https://github.com/medusajs/medusa/pull/6101#discussion_r1457251062 feel free to jump on it

Reply:
so the convention have been validated, only the relation can't have a default initializer when they are nullable

Reply:
sounds good, I've updated the PR already

Reply:
you are too quick man 🚀